pkg/pool-stable/test/StablePool.test.ts

```diff
@@ -294,13 +308,22 @@ describe('StablePool', function () {
             expect(currentBptBalance.sub(previousBptBalance)).to.be.equalWithError(expectedBptOut, 0.0001);
           });
 
-          it('can tell how much BPT it will give in return', async () => {
+          async function joinGivenIn(): Promise<void> {
             const minimumBptOut = pct(expectedBptOut, 0.99);
 
             const result = await pool.queryJoinGivenIn({ amountsIn, minimumBptOut });
 
             expect(result.amountsIn).to.deep.equal(amountsIn);
             expect(result.bptOut).to.be.equalWithError(expectedBptOut, 0.0001);
+          }
+
+          it('can tell how much BPT it will give in return', async () => {
+            await joinGivenIn();
+          });
+
+          it('can tell how much BPT it will give in return (in recovery mode)', async () => {
+            await enterRecoveryMode(pool);
+            await joinGivenIn();
```

Comment:
This is, however, out of scope. BasePool allows for regular joins and exits outside of recovery mode, there's nothing to test here.

Reply:
I didn't add these tests - it was like that before. I just added the "in recovery mode" rerun of it.

Reply:
That's what I mean - we don't need to test that the joins and exits work properly in recovery mode because that is part of the regular behavior, there's nothing special about it. But if you want to add it for completeness, I'd do it in the way suggested above.